pkg/volume/host_path/host_path.go

```diff
@@ -99,14 +100,22 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
 
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path = filepath.Join("/rootfs", path)
```

Comment:
@dixudx If the kubelet already depends on `--privileged` and having `nsenter` available. `pid=host` isn't needed since `containerized` currently implies a `/rootfs` mount, which contains `/rootfs/proc`.

The kubelet as a whole depends on nsenter though as port-forwarding is done via `nsenter` + `socat`.

Also, if you look at https://github.com/kubernetes/kubernetes/blob/v1.7.2/pkg/util/io/writer.go#L54-L59 and https://github.com/kubernetes/kubernetes/blob/v1.7.2/pkg/util/mount/nsenter_mount.go you'll again see those assumptions encoded for `--containerized`

If the intent of the code is to deal with `--containerized`, then those assumptions are all already existent.
I agree it's fragile and messy, but that's what `--containerized` is right now.